qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/AbstractRestartUpgradeRestTestCase.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.neuralsearch.bwc;
+
+import java.util.Locale;
+import java.util.Optional;
+import org.junit.Before;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+import static org.opensearch.neuralsearch.TestUtils.CLIENT_TIMEOUT_VALUE;
+import static org.opensearch.neuralsearch.TestUtils.RESTART_UPGRADE_OLD_CLUSTER;
+import static org.opensearch.neuralsearch.TestUtils.BWC_VERSION;
+import static org.opensearch.neuralsearch.TestUtils.NEURAL_SEARCH_BWC_PREFIX;
+import org.opensearch.test.rest.OpenSearchRestTestCase;
+
+public abstract class AbstractRestartUpgradeRestTestCase extends BaseNeuralSearchIT {
+    protected static String testIndex;
```

Comment:
does this need to be static?

Reply:
Yes it should be static. As the name for testIndex is set once and need not to be changed. If any issues comes then it is easy to debug. Otherwise changing it to non static will again and again require to create object and assign a new value in every test.

Reply:
With the @Before annotation, it will be reset for each test. So Im not sure it needs to be static. I dont think there is any benefit.